src/AWS.Deploy.CLI/Commands/GenerateDeploymentProjectCommand.cs

```diff
@@ -0,0 +1,218 @@
+                    "future redeployments to previously deployed applications. " + Environment.NewLine + Environment.NewLine +
+                    "Do you still want to continue?";
+
+                _toolInteractiveService.WriteLine();
+                var yesNoResult = _consoleUtilities.AskYesNoQuestion(userPrompt, YesNo.Yes);
+
+                if (yesNoResult == YesNo.No)
+                {
+                    if (newDirectoryCreated)
+                        _directoryManager.Delete(saveCdkDirectoryPath);
+                    return;
+                }
+            }
+
+            await _cdkProjectHandler.CreateCdkProjectForDeployment(selectedRecommendation, _session, saveCdkDirectoryPath);
+            await GenerateDeploymentRecipeSnapShot(selectedRecommendation, saveCdkDirectoryPath, projectDisplayName);
+
+            var saveCdkDirectoryInfo = _directoryManager.GetDirectoryInfo(saveCdkDirectoryPath);
+            _toolInteractiveService.WriteLine();
+            _toolInteractiveService.WriteLine($"The CDK deployment project is saved at: {saveCdkDirectoryInfo.FullName}");
+        }
+
+        /// <summary>
+        /// This method generates the appropriate recommendations for the target deployment project.
+        /// </summary>
+        /// <param name="orchestrator"><see cref="Orchestrator"/></param>
+        /// <returns>A list of <see cref="Recommendation"/></returns>
+        private async Task<List<Recommendation>> GenerateDeploymentRecommendations(Orchestrator orchestrator)
+        {
+            var recommendations = await orchestrator.GenerateDeploymentRecommendations(forDeployment: false);
+            if (recommendations.Count == 0)
+            {
+                throw new FailedToGenerateAnyRecommendations("The project you are trying to deploy is currently not supported.");
+            }
+            return recommendations;
+        }
+
+        /// <summary>
+        /// This method takes the path to the target deployment project and creates
+        /// a default save directory inside the parent folder of the current directory.
+        /// For example:
+        /// Target project directory - C:\Codebase\MyWebApp
+        /// Generated default save directory - C:\Codebase\MyWebApp.Deployment If the save directory already exists, then a suffix number is attached.
+        /// </summary>
+        /// <returns>The defaukt save directory path.</returns>
+        private string GenerateDefaultSaveDirectoryPath()
+        {
+            var applicatonDirectoryFullPath = _directoryManager.GetDirectoryInfo(_targetApplicationFullPath).Parent.FullName;
+            var saveCdkDirectoryFullPath = applicatonDirectoryFullPath + ".Deployment";
+
+            var suffixNumber = 0;
+            while (_directoryManager.Exists(saveCdkDirectoryFullPath))
+                saveCdkDirectoryFullPath = applicatonDirectoryFullPath + $".Deployment{++suffixNumber}";
```

Comment:
Can you confirm with @ik130 the naming?

Reply:
Changed the suffix to `CDK` instead of `.Deployment`. This change came from @ik130.